lib/webtask.js

```diff
@@ -797,3 +879,27 @@ function parse_ten(ten) {
 function normalize_host(host) {
     return (host ? host.toLowerCase().replace(/\:/g, '-') : 'global');
 }
+
+function wants_html(req) {
```

Comment:
My thinking was:

If the client says it `Accepts` HTML, then the client can use the auth dialogue. Whether it *prefers* HTML at them moment seems irrelevant to the reply, since the alternative is a 403 and nothing anyway.

But I do agree this still isn't ideal solution, though since it's overridable it doesn't seem like a dealbreaker to me.